pkg/kubectl/cmd/apply.go

```diff
@@ -159,6 +174,26 @@ func NewCmdApply(baseName string, f cmdutil.Factory, out, errOut io.Writer) *cob
 	return cmd
 }
 
+func (o *ApplyOptions) Complete(cmd *cobra.Command) error {
+	o.DryRun = cmdutil.GetDryRunFlag(cmd)
+
+	// allow for a success message operation to be specified at print time
+	o.PrintObj = func(obj runtime.Object, operation string) error {
```

Comment:
Icky, but better than we have I suppose.

`func(operation string) printers.ResourcePrinterFunc` though to make it clear what is happening.